Bash conditionals loops

drelliche opened the pull request.

drelliche commented:
Once the language in Bash 102 for finding file paths is finalized (PR https://github.com/arcus/education_modules/pull/152), this module needs to be updated and will then be ready for review.

leemc-data-ed commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+
+</div>
+
+The statement `[ 5 -eq 05 ]` is checking if the numbers `5` and `05` are equal to each other. The command `echo "integer equality"` only runs if the numbers are equal. If you change the `05` to `04`, this code will not return any output.
```
Thinking that maybe a highlight box would be useful here, to really highlight that integer equality uses `-eq` and not `=`, since that's probably not intuitive.

drelliche replied:
Added a sentence about `=` and mentioned that it will be addressed more later in the module.

leemc-data-ed commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+
+</div>
+
+You can have as many `elif` and `then` pairs as you want, but as soon as one is statement is true, the code will do that action stop. It will not continue to check the remaining statements. Try to think about what this code will output, then run it and check if you were right:
```
```suggestion
You can have as many `elif` and `then` pairs as you want, but as soon as one is statement is true, the code will do that action **and then stop**. It will not continue to check the remaining statements! Try to think about what this code will output, then run it and check if you were right:
```

leemc-data-ed commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+
+## Additional Resources
+
+Software Carpentry has a series of lessons on the [Unix Shell](https://swcarpentry.github.io/shell-novice/), including one specifically on [loops](https://swcarpentry.github.io/shell-novice/05-loop/index.html).
```
We should try to come up with some other resources, since this is tricky. Maybe some cheatsheets... it might be cool to also have a link to something about creating flowcharts for loops (mostly the ones I've found are for Python, but I can keep looking)

drelliche replied:
I found some more to add in here, two cheat sheets, a cute video, and programming language agnostic flow charts!

drelliche commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+
+**Loops**
+
+A **loop** is a bit of code that allows you to run the same command again and again. The loop gives instructions for how many times to run your code, and if it requires input, what that input should be.  With loops, you get to write your command once and then tell it to run as many times as you want!
```
Great thought! I added a sentence at the end of the Iteration paragraph to that effect!

drelliche commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+
+This is where conditional statements come in! A conditional evaluates if a statement is true, and if it is true, executes a command. The most basic conditional statements take the form "if/then."
+
+"**If** the file is empty, **then** delete the file."
```
Great idea about using a code block!

drelliche commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+* how to recognize conditionals in Bash
+* how to make small changes to existing code.
+
+If you find that you need to use conditionals a lot in Bash, you will probably settle on a favorite cheat sheet, of which there are many. (links here and in additional resources)
```
sure do 😬

drelliche commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+
+Bash is an older language and can be used to on all sorts of computers with different operating systems to interface with file and directories. This versatility means that Bash shows up in a lot of places. Unfortunately that does not mean that Bash is easy to understand. In fact the opposite is true.
+
+One of Bash's worst properties is that, unlike most programming languages which think of "true" statements as `1` and "false" statements as `0`, in Bash a "true" statement is assigned the numerical value `0` (except when evaluating mathematical expressions inside of double parentheses).
```
Great suggestion! "Worst" is probably a personal opinion of mine, but it is VERY strongly held 😄 

Still, I've toned down some of my rhetoric and added a pretty great stack exchange explanation of the logic of it.

drelliche commented on bash_conditionals_loops/bash_conditionals_loops.md:
```diff
@@ -0,0 +1,620 @@
+3. Once the Zip file has downloaded, un-zipping it will create a folder titled `learning_bash-main`. Depending on your computer's operating system, you may be able to un-zip the folder by double clicking on it, or may need to right click on it a select "Extract All." This may create an identically named folder inside `learning_bash-main` that contains all of the individual files.
+4. [Find out the file path](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/directories_and_file_paths/directories_and_file_paths.md#6) (location on your computer) of the new folder `learning_bash-main` and navigate there in your command line interface.
+
+ <div class = "help">
```
fixed

drelliche commented on bash_conditionals_loops/bash_conditionals_loops.md:
````diff
@@ -0,0 +1,620 @@
+
+The word `in` after defining the variable is important, and the command `do` comes before `done`. The correct code is:
+```
+for animal in giant_squid elephant albatross
````
fixed!

drelliche commented:
@leemc-data-ed I fixed the rendering issues, added a lot more links, and incorporated all of your extremely helpful comments! 

Let me know what you think :)

leemc-data-ed reviewed (approved):
Lgtm! Approved!
